File: java/space-age/SpaceAge.java
class SpaceAge {
    private double seconds;

    SpaceAge(double seconds) {
        this.seconds = seconds;
    }

    double getSeconds() {
        return seconds;
    }

    double onEarth()   { return seconds / 31557600.0; }
    double onMercury() { return seconds / 7600543.81992; }
    double onVenus()   { return seconds / 19414149.052176; }
    double onMars()    { return seconds / 59354032.69008; }
    double onJupiter() { return seconds / 374355659.124; }
    double onSaturn()  { return seconds / 929292362.8848; }
    double onUranus()  { return seconds / 2651370019.3296; }
    double onNeptune() { return seconds / 5200418560.032001; }

}

File: ruby/space-age/space_age.rb
class SpaceAge

  def initialize(seconds)
    @seconds = seconds
  end

  @@planets_to_years = { # orbital period of each planet
    on_earth: 31557600.0,
    on_mercury: 7600543.81992,
    on_venus: 19414149.052176,
    on_mars: 59354032.69008,
    on_jupiter: 374355659.124,
    on_saturn: 929292362.8848,
    on_uranus: 2651370019.3296,
    on_neptune: 5200418560.032001,
  }

  def method_missing(m, *args, &block)
    return (@seconds / @@planets_to_years[m]) if @@planets_to_years.has_key? m

    raise NoMethodError, "undefined method `#{m}' for #{self.inspect}"
  end
end
